elytron/src/test/java/org/wildfly/extension/elytron/SyslogAuditLogTestCase.java

```diff
@@ -0,0 +1,334 @@
+     * of RFC5424 by verifying the connection message
+     */
+    @Test
+    public void testDefaultSyslogFormat() throws Exception {
+        assertSuccess(services.executeOperation(udpOperation));
+        verifyUdpMessage(RFC5424_SYSLOG_MESSAGE);
+    }
+
+    /**
+     * Tests that the syslog configuration operation fails if an invalid RFC format is used
+     */
+    @Test
+    public void testWrongSyslogFormat() {
+        udpOperation.get(ElytronDescriptionConstants.SYSLOG_FORMAT).set(BAD_RFC_STRING);
+        ModelNode response = services.executeOperation(udpOperation);
+        assertCorrectError(response, new String[] {"WFLYCTL0158", "RFC1"});
+        assertFailed(response);
+    }
+
+    /**
+     * Tests that the syslog configuration operation fails if an invalid reconnect-attempts number is used
+     */
+    @Test
+    public void testBadIntegerReconnectAttempts() {
+        udpOperation.get(ElytronDescriptionConstants.RECONNECT_ATTEMPTS).set(BAD_RECONNECT_NUMBER);
+        ModelNode response = services.executeOperation(udpOperation);
+        assertCorrectError(response, new String[] {"WFLYCTL0117", Integer.toString(BAD_RECONNECT_NUMBER)});
+        assertFailed(response);
+    }
+
+    /**
+     * Tests that the server does not attempt to resend the message
+     */
+    @Test
+    public void testZeroReconnectAttemptsBadHost() {
+        // Windows Server can have an issue echoing back an error, causing the test to fail by not seeing a failure sending
+        Assume.assumeFalse("Test does not run on Windows Server", SystemUtils.OS_NAME.contains("Windows Server"));
+        badHostUdpOperation.get(ElytronDescriptionConstants.RECONNECT_ATTEMPTS).set(0);
+        ModelNode response = services.executeOperation(badHostUdpOperation);
+        assertCorrectError(response, new String[] {"ELY12001", "0"});
+        assertFailed(response);
+    }
+
+    /**
+     * Tests that the server reattempts to send the message {@link SyslogAuditLogTestCase#RECONNECT_NUMBER} times
+     */
+    @Test
+    public void testNumberedReconnectAttemptsBadHost() {
+        // Windows Server can have an issue echoing back an error, causing the test to fail by not seeing a failure sending
+        Assume.assumeFalse("Test does not run on Windows Server", SystemUtils.OS_NAME.contains("Windows Server"));
+        badHostUdpOperation.get(ElytronDescriptionConstants.RECONNECT_ATTEMPTS).set(RECONNECT_NUMBER);
+        ModelNode response = services.executeOperation(badHostUdpOperation);
+        assertCorrectError(response, new String[] {"ELY12001", Integer.toString(RECONNECT_NUMBER)});
+        assertFailed(response);
+    }
+
+    /**
+     * Tests that the server continuously attempts to resend the message for {@link SyslogAuditLogTestCase#RECONNECT_TIMEOUT} milliseconds
+     */
+    @Test
+    public void testInfiniteReconnectAttemptsBadHost() throws Exception {
```

Comment:
I'm not sure I understand this test. It looks like it only tests that the operation fails.

Reply:
It's to test that it doesn't stop resending the message before the interrupt from the parent thread happens, since if the it either thinks it successfully sent the message or stops sending it without the parent thread interrupting then the ModelNode response will not be cancelled and the test will fail.